services/viva-ms/lambdas/createVivaCase.js

```diff
@@ -276,27 +299,33 @@ async function getUser(PK) {
   return getResult.Item;
 }
 
-async function getFormTemplates(forms) {
-  const formTemplates = {};
-
-  for (const formId of Object.keys(forms)) {
-    const formGetParams = {
-      TableName: config.forms.tableName,
-      Key: {
-        PK: `FORM#${formId}`,
-      },
-    };
-
-    const [getError, getResult] = await to(dynamoDB.call('get', formGetParams));
-    if (getError) {
-      console.error('(viva-ms) DynamoDb query on forms table failed', getError);
-      continue;
-    }
+async function getFormTemplates(formIds) {
+  const [getError, forms] = await to(
+    Promise.all(
+      formIds.map(formId => {
+        const formGetParams = {
+          TableName: config.forms.tableName,
+          Key: {
+            PK: `FORM#${formId}`,
+          },
+        };
+        return dynamoDB.call('get', formGetParams);
+      })
+    )
+  );
 
-    formTemplates[formId] = getResult.Item;
+  if (getError) {
+    console.error('(viva-ms) DynamoDb query on forms table failed', getError);
+    throw getError;
   }
 
-  return formTemplates;
+  return forms.reduce(
+    (templates, form) => ({
+      ...templates,
+      [form.Item.id]: form.Item,
+    }),
+    {}
+  );
```

Comment:
We try to place the return value in descriptive variables before we return.
Suggestion:
formTemplates = forms.reduce...
return formTemplates

One benefit is that you can easy log the variable formTemplates. Good when debugging

The name can certainly be better. Honestly, I have a hard time clearly seeing what the result will be? :)